pylzy/lzy/env/provisioning/score.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from typing import Callable
+
+from lzy.types import VmResources
+
+ScoreFunctionType = Callable[[VmResources, VmResources], float]
+
+DEFAULT_WEIGHTS = {
+    'gpu': 100.0,
+    'cpu': 20.0,
+    'ram': 1.0
+}
+
+
+def maximum_score_function(requested: VmResources, spec: VmResources) -> float:
+    """Score function that scores big machines the max.
+    Use it when you need the best spec available.
+    """
+
+    d_gpu = spec.gpu_count - requested.gpu_count
+    d_cpu = spec.cpu_count - requested.cpu_count
+    d_ram = spec.ram_size_gb - requested.ram_size_gb
+
+    return d_gpu * DEFAULT_WEIGHTS['gpu'] + d_cpu * DEFAULT_WEIGHTS['cpu'] + d_ram * DEFAULT_WEIGHTS['ram']
```

Comment:
`d_gpu` and other summands can be negative which made them unapplicable
but with this logic, we can have not enough requested resource, but a lot of others (i.e. no GPU but a lot of CPU while requested some GPU), which will made this spec a preferred choice

Reply:
may be add TODO-s?